### PR TITLE
8339548: GHA: RISC-V: Use Debian snapshot archive for bootstrap

### PR DESCRIPTION
**Pull request by RealFYang**
Debian "sid" or "unstable" (https://httpredir.debian.org/debian) that we use for debootstrapping RISC-V breaks very often. Currently, the GHA linux-cross-build for RISC-V would not continue and is simply skipped when this debootstrap for "sid" repos fails. (See [JDK-8326960](https://bugs.openjdk.org/browse/JDK-8326960) for more details). This is affecting GHA linux-cross-build for RISC-V for quite some time. As a result, we failed to catch some early build issues.

But I don't think we need to catch up with the latest Debian "unstable" for our GHA verification purpose. So one way would be using some older but working Debian shapshot [1] [2] for our purpose. I find the most recent usable shapshot is https://snapshot.debian.org/archive/debian/20240228T034848Z/. We can switch back to more stable Debian repo once it graduates.

[1] https://snapshot.debian.org/
[2] https://lists.debian.org/debian-snapshot/

Testing:
- [x] GHA

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8339548](https://bugs.openjdk.org/browse/JDK-8339548): GHA: RISC-V: Use Debian snapshot archive for bootstrap (**Bug** - P4)

### Reviewers
 * [Aleksey Shipilev](https://openjdk.org/census#shade) (@shipilev - **Reviewer**)
 * [Erik Joelsson](https://openjdk.org/census#erikj) (@erikj79 - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/20853/head:pull/20853` \
`$ git checkout pull/20853`

Update a local copy of the PR: \
`$ git checkout pull/20853` \
`$ git pull https://git.openjdk.org/jdk.git pull/20853/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 20853`

View PR using the GUI difftool: \
`$ git pr show -t 20853`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/20853.diff">https://git.openjdk.org/jdk/pull/20853.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/20853#issuecomment-2329354654)

### CODE REVIEW

**Comment by RealFYang** [comment]
@shipilev @erikj79 : Please take a look. Thanks.

**Review by shipilev — APPROVED** [review]
I don't mind doing this, especially given that RISC-V job would "pass" cleanly if this snapshot does not work. It is on you to make sure it does work :)

**Comment by RealFYang** [comment]
OK. Great. I see GHAs are now clean with this change. 
I will integrate after I get a second review from build/GHA maintainers. Maybe @erikj79 ? Thanks.

**Review by erikj79 — APPROVED** [review]

**Comment by erikj79** [comment]
> @erikj79 : Please take a look. Thanks.

> Maybe @erikj79 ? Thanks.

I generally skip most GHA related PRs as there are plenty of other reviewers on this mailing list with deeper GHA knowledge than I have, and it's not a focus for my current role at my employer. This patch was however small enough so I was comfortable approving it. 

Tagging me, or someone else, specifically in a PR, requesting a review with the intent of getting it quicker, will likely just have the opposite effect as it may deter other people from jumping in to review, thinking you already have some kind of agreement with the person being tagged.

**Comment by RealFYang** [comment]
@erikj79: Thanks for having a look. The reason why I tagged you is that I was thinking that you (as a reviewer of the previous https://github.com/openjdk/jdk/pull/18435) might be more familar with the background of this GHA issue. Apologies anyway :-)

**Comment by RealFYang** [comment]
I see GHA linux-cross-build job for RISC-V is still skipped due to broken sysroot. Let's get this integrated to reactivate this job.
/integrate

**Pull request closed without merging**
